### PR TITLE
Move PhoneInput, Spinner, and CountrySelect @automattic/components

### PR DESCRIPTION
**Pull request by senadir**
#### Changes proposed in this Pull Request
_This PR is 1/3 of moving the components, the first PR will only move files and update dependencies._

* This PR tries to move `PhoneInput` and its peer dependencies `Spinner` and `FormCountrySelect` to `packages/components` aka `@automattic/components` so other projects (like Woo Blocks) can use them, while this works, for now, there are things that need to be reviewed:

- What should we do with `i18-calypso`? @blowery suggested in p1576091383223700-slack-C02DQP0FP that I just removed and supply it as a prop.
- Does it make sense to publish flags or should we reconsider how to use it?
- There were some debug calls in `country-flag` that I took liberty in removing so I don’t add extra things to the build, that should be reviewed I guess.
- Some linter issues here and there, one, in particular, is using `setState` inside `didComponentUpdate` in `country-flag.jsx#L29`
- Docs needs some updates as well, remove linking to devdocs as well.
- I didn't yet remove the components from `client/components`, figured until we're sure I didnt break anything till I do that.
#### Testing instructions
- in ` client/components/forms/form-phone-media-input/index.jsx` update the call to `PhoneInput` 
from: 
```js
import PhoneInput from 'components/phone-input';
```
to:
```js
import { PhoneInput } from '@automattic/components';
```
- Fire Calypso and add a domain to your cart and try to checkout

Alternatively, try this with `devdocs`, so in `client/components/forms/docs/example.jsx` update the import.
- visit `http://calypso.localhost:3000/devdocs/design/form-fields`

### CODE REVIEW

File: packages/components/package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
 	"dependencies": {
 		"classnames": "^2.2.6",
 		"enzyme": "^3.10.0",
+		"flag-icon-css": "3.4.5",
```
Comment (blowery) [inline packages/components/package.json]: this should be a caret rev 
```suggestion
		"flag-icon-css": "^3.4.5",
```

File: packages/components/src/spinner/style.scss [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+.spinner {
+	display: flex;
+	align-items: center;
+}
+
+.spinner__outer, .spinner__inner {
+	margin: auto;
+	box-sizing: border-box;
+  border: 0.1em solid transparent;
```
Comment (blowery) [inline packages/components/src/spinner/style.scss]:
```suggestion
	border: 0.1em solid transparent;
```

File: packages/components/src/spinner/style.scss [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	animation-name: rotate-spinner;
+}
+
+.spinner__outer {
+	border-top-color: var( --color-accent );
+}
+
+.spinner__inner {
+	width: 100%;
+  height: 100%;
```
Comment (blowery) [inline packages/components/src/spinner/style.scss]:
```suggestion
	height: 100%;
```

File: packages/components/src/spinner/style.scss [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+	border-top-color: var( --color-accent );
+  border-right-color: var( --color-accent );
```
Comment (blowery) [inline packages/components/src/spinner/style.scss]:
```suggestion
	border-right-color: var( --color-accent );
```

File: packages/components/src/spinner/style.scss [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  opacity: 0.4;
```
Comment (blowery) [inline packages/components/src/spinner/style.scss]:
```suggestion
	opacity: 0.4;
```

**Comment by senadir** [comment]
@blowery I know this PR is huge to review since everything seems to have been changed, I sorted my changes commit by commit so you can see exactly what changed (initial commit was moving files, what followed was some simple path refactoring)

what else do you think is left for this PR to be driven further?

**Comment by sirbrillig** [comment]
You weren't asking me, but my suggestion would be to split this into several PRs:

1. Add the components to `@automattic/components`.
2. Switch the existing code to use the new package.
3. Remove the (now unused) old components.

That way the second PR would be relatively small (it would only be changes) and may be easier to review. The first and third PRs might be much larger, but they should have no effects so they are less risky.

**Comment by blowery** [comment]
Thanks @sirbrillig, that's exactly what I would have said. :)

**Pull request closed without merging**